acloud/data_source_cloud_account.go

```diff
@@ -48,8 +46,9 @@ func dataSourceCloudAccount() *schema.Resource {
 }
 
 func dataCloudAccountRead(ctx context.Context, d *schema.ResourceData, m interface{}) diag.Diagnostics {
-	client := m.(acloudapi.Client)
-	org := d.Get("organisation").(string)
+	provider := m.(ConfiguredProvider)
```

Comment:
I would suggest creating a helper method for this.

In the helper method you should include error handling. What if `m` is not a `ConfiguredProvider`.

Reply:
Since the error handling is an internal error, you could possible use `panic("m must be ConfiguredProvider")`.

This way all usages of the new helper method do not have to do the error handling.